src/com/oltpbenchmark/benchmarks/auctionmark/AuctionMarkProfile.java

```diff
@@ -534,266 +525,273 @@ public synchronized Timestamp updateAndGetLastCloseAuctionsTime() {
             for (ItemInfo itemInfo : items) {
                 this.addItemToProperQueue(itemInfo, currentTime);
-            } // FOR
+            }
         }
-        
+
         if (LOG.isDebugEnabled()) {
-            Map<ItemStatus, Integer> m = new HashMap<ItemStatus, Integer>();
+            Map<ItemStatus, Integer> m = new HashMap<>();
             m.put(ItemStatus.OPEN, this.items_available.size());
             m.put(ItemStatus.ENDING_SOON, this.items_endingSoon.size());
             m.put(ItemStatus.WAITING_FOR_PURCHASE, this.items_waitingForPurchase.size());
             m.put(ItemStatus.CLOSED, this.items_completed.size());
             LOG.debug(String.format("Updated Item Queues [%s]:\n%s",
-                                    currentTime, StringUtil.formatMaps(m)));
+                    currentTime, StringUtil.formatMaps(m)));
         }
     }
-    
+
     public ItemStatus addItemToProperQueue(ItemInfo itemInfo, boolean is_loader) {
         // Calculate how much time is left for this auction
         Timestamp baseTime = (is_loader ? this.getLoaderStartTime() : this.getCurrentTime());
-        assert(itemInfo.endDate != null);
-        assert(baseTime != null) : "is_loader=" + is_loader;
+
+
         return addItemToProperQueue(itemInfo, baseTime);
     }
-        
+
     private ItemStatus addItemToProperQueue(ItemInfo itemInfo, Timestamp baseTime) {
         // Always check whether we even want it for this client
         // The loader's profile and the cache profile will always have a negative client_id,
         // which means that we always want to keep it
         if (this.client_id != -1) {
-            if (this.userIdGenerator == null) this.initializeUserIdGenerator(this.client_id);
-            if (this.userIdGenerator.checkClient(itemInfo.getSellerId()) == false) {
-                return (null);
+            if (this.userIdGenerator == null) {
+                this.initializeUserIdGenerator(this.client_id);
             }
-        }
-        
-        long remaining = itemInfo.endDate.getTime() - baseTime.getTime();
-        ItemStatus new_status = (itemInfo.status != null ? itemInfo.status : ItemStatus.OPEN); 
-        // Already ended
-        if (remaining <= AuctionMarkConstants.ITEM_ALREADY_ENDED) {
-            if (itemInfo.numBids > 0 && itemInfo.status != ItemStatus.CLOSED) {
-                new_status = ItemStatus.WAITING_FOR_PURCHASE;
-            } else {
-                new_status = ItemStatus.CLOSED;
+            if (!this.userIdGenerator.checkClient(itemInfo.getSellerId())) {
+                return (null);
             }
         }
-        // About to end soon
-        else if (remaining < AuctionMarkConstants.ITEM_ENDING_SOON) {
+
+        long remaining = itemInfo.getEndDate().getTime() - baseTime.getTime();
+
+
+        ItemStatus existingStatus = itemInfo.getStatus();
+        ItemStatus new_status = (existingStatus != null ? existingStatus : ItemStatus.OPEN);
+
+        if (remaining <= 0) {
```

Comment:
TODO check equivalence of conditions

Reply:
I _think_ the new code is more correct, even if not strictly equivalent around some odd cases.